spec/controller_extension_spec.rb

```diff
@@ -0,0 +1,79 @@
+# spec/readyset/controller_extension_spec.rb
+
+require 'rails_helper'
+require 'ready_set/controller_extension'
+
+RSpec.describe ReadySet::ControllerExtension, type: :controller do
+  # Global Set-up
+  controller(ActionController::Base) do
+    # NOTE: Not required as it's setup by the ReadySet Railtie
+    include ReadySet::ControllerExtension
+
+    # Main point-of-interest in our fake controller
+    # This line specifies that these queries will be re-routed
+    route_to_readyset only: [:index, :show]
+
+    def index
+      @posts = Post.where(active: true)
+      render plain: 'Index'
+    end
+
+    def show
+      @post = Post.find(params[:id])
+      render plain: 'Show'
+    end
+
+    def create
+      Post.create(params[:post])
+      render plain: 'Create'
+    end
+  end
+
+  before do
+    # Need a way to clean this up
+    routes.draw do
+      get 'index' => 'anonymous#index'
+      get 'show/:id' => 'anonymous#show'
+      post 'create' => 'anonymous#create'
+    end
+
+    # Our fake queries
+    stub_const('Post', Class.new)
+    allow(Post).to receive(:where).and_return([])
+    allow(Post).to receive(:find).and_return(nil)
+    allow(Post).to receive(:create)
+  end
+
+  describe '#route_to_readyset' do
+    # Lacks full coverage of possible #around_action parameters, but gets the point across
```

Comment:
I think it's okay that we don't test all of the `around_action` params, since we just pass those params to `around_action` anyway. I do think it's worth adding a test case here to assert that we do actually pass those options un-altered to `around_action`

Reply:
Gotcha gotcha.